src/public/publicAPIs.ts

```diff
@@ -312,4 +312,22 @@ export function navigateToTab(tabInstance: TabInstance, onComplete?: (status: bo
 
   const errorMessage = "Invalid internalTabInstanceId and/or channelId were/was provided";
   GlobalVars.callbacks[messageId] = onComplete ? onComplete : getGenericOnCompleteHandler(errorMessage);
-}
\ No newline at end of file
+}
+
+/**
+ * Sends query to bot in order to retrieve data.
+ * @param botRequest query to send to bot.
+ * @param callback callback to invoke when data is retrieved from bot
+ */
+export function getBotData(botRequest: BotAPIRequest, callback?: (botResponse: any) => void): void { // void for now
```

Comment:
rename this to executeBotQuery

Reply:
Can we make it more generic, say sendBotRequest or sendBotMessage()? I can see this being used for more than queries.

Reply:
sendBotRequest ?